code/modules/mob/living/basic/bots/vibebot/vibebot_action.dm

```diff
@@ -0,0 +1,65 @@
+/**
+ * Vibebot's vibe ability
+ *
+ * Given to vibebots so sentient ones can change/reset thier colors at will.
+ */
+/datum/action/innate/vibe
+	name = "Vibe"
+	desc = "LMB: Change vibe color. RMB: Reset vibe color."
+	button_icon = 'icons/mob/actions/actions_minor_antag.dmi'
+	button_icon_state = "funk"
+
+/datum/action/innate/vibe/Grant(mob/grant_to)
+	. = ..()
+	RegisterSignal(grant_to, COMSIG_BOT_TURNED_ON, PROC_REF(activate_lights))
+	RegisterSignal(grant_to, COMSIG_CHANGE_VIBEBOT_COLOR, PROC_REF(change_color))
+	RegisterSignal(grant_to, COMSIG_BOT_TURNED_OFF, PROC_REF(turn_off_lights))
+
+/datum/action/innate/vibe/Remove(mob/removed_from)
+	UnregisterSignal(removed_from, list(COMSIG_BOT_TURNED_ON, COMSIG_BOT_TURNED_OFF))
```

Comment:
You're not unregistering ``COMSIG_CHANGE_VIBEBOT_COLOR``?

Reply:
whoops i forgot